File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/adapter/Traveler.java
package adapter;

public class Traveler {
    public static void main(String[] args) {
        // Create electronic device
        ElectronicDevice device = new ElectronicDevice();

        // Create universal adapter
        UniversalAdapter adapter = new UniversalAdapter(device);

        // Connect device to electrical outlet and charge
        adapter.request();
    }
}

File: code/csharp/DesignPatterns/adapter/Traveler.cs
using System;

public class Traveler
{
    public static void AdapterExample()
    {
        // Create electronic device
        MobilePhone device = new MobilePhone();

        // Create universal adapter
        UniversalAdapter adapter = new UniversalAdapter(device);

        // Connect device to electrical outlet and charge
        adapter.Request();
    }
}
